internal/api/authz/authorization.go

```diff
@@ -125,3 +128,47 @@ func GetAllPermissionCtxIDs(perms []string) []string {
 	}
 	return ctxIDs
 }
+
+func addGetSystemUserRolesFuncToCtx(ctx context.Context, ctxData CtxData) context.Context {
+	if len(ctxData.SystemMemberships) != 0 {
+		ctx = context.WithValue(ctx, systemUserRolesFuncKey, func() func(ctx context.Context) ([]string, error) {
+			var roles []string
+			return func(ctx context.Context) ([]string, error) {
+				if roles != nil {
+					return roles, nil
+				}
+				var err error
+				roles, err = getSystemUserRoles(ctx)
+				return roles, err
+			}
+		}())
+	}
+	return ctx
+}
+
+func GetSystemUserRoles(ctx context.Context) ([]string, error) {
+	getSystemUserRolesFuncValue := ctx.Value(systemUserRolesFuncKey)
+	if getSystemUserRolesFuncValue == nil {
+		return nil, nil
+	}
+	getSystemUserRolesFunc, ok := getSystemUserRolesFuncValue.(func(context.Context) ([]string, error))
+	if !ok {
+		return nil, errors.New("unable to obtain systems role func")
```

Comment:
I'm wondering if it's really needed to return an error here. What will fail if there are no system membership entries or function? Can't we just return a empty value and only log the error of type-mismatch? Technically type mis-match would be a bug and shouldn't really get exposed on the API.